Assets/Scripts/Player/PlayerLook.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerLook : MonoBehaviour
+{
+    [SerializeField] private Transform m_camera;
+    [SerializeField] private Transform m_player;
+    [Space]
+    [SerializeField] private float m_mouseSensitivity = 50f;
+
+    public bool m_hideLockCamera = true;
+
+    private float m_xRotation;
+
+    private void Start()
+    {
+        CursorCheck();
+    }
+
+    private void FixedUpdate()
+    {
+        MouseControl();
+    }
+
+    private void MouseControl()
+    {
+        float mouseX = Input.GetAxis("Mouse X") * Time.deltaTime * m_mouseSensitivity;
+        float mouseY = Input.GetAxis("Mouse Y") * Time.deltaTime * m_mouseSensitivity;
+
+        m_xRotation -= mouseY;
+        m_xRotation = Mathf.Clamp(m_xRotation, -90f, 90f);
+
+        m_camera.localRotation = Quaternion.Euler(m_xRotation, 0f, 0f);
+        m_player.Rotate(Vector3.up * mouseX);
+    }
+
+    private void CursorCheck()
```

Comment:
Rename CursorCheck as well, by reading it, I have no idea what it does. We could actually use a util for the cursor, so we can disable and enable the cursor from anywhere and do other useful things without having to write the block of code that does it all.